tremor-cli/src/target_process.rs

```diff
@@ -131,6 +131,14 @@ impl TargetProcess {
             };
             buf.canonicalize()?
         };
+        // env var replacement
+        let mut args: Box<dyn Iterator<Item = String>> = Box::new(args.iter().cloned());
+        for (var, value) in env {
+            args = Box::new(args.map(|s| s.replace(&format!("${}", var.clone()), value)));
```

Comment:
I don't think we need the clone here, as it's in a  format it can be borrowed.

Reply:
I wouldn't have added it, if the borrow checker didn't tell me it was necessary. The thing is that this is an iterator, which only lazily applies the closure to the base-iterator, so `var` needs to be cloned to outlive the containing loop iteration.